app/src/main/java/org/cru/godtools/activity/BasePlatformActivity.java

```diff
@@ -120,15 +121,23 @@ protected void onSetupActionBar() {
     @Override
     protected void onStart() {
         super.onStart();
-        startLanguagesChangeListener();
+        startSettingsChangeListener();
         mEventBus.register(this);
         loadLanguages(false);
         updateNavigationDrawerMenu();
     }
 
-    @MainThread
-    @Subscribe(threadMode = ThreadMode.MAIN)
-    public void onTheKeyEvent(@NonNull final TheKeyEvent event) {
+    @CallSuper
+    public void onSettingsUpdated(@Nullable final SharedPreferences preferences, @Nullable final String key) {
```

Comment:
I made this overridable so that the signup banner UI can be refreshed when the login feature preference changes.